Error: Cannot find module '@storybook/global'

Hi, I tried to integrate your addon using the context described below:
- Storybook 7.6.3 (also tried using 7.5)
- Angular 16.2.10

But when I start Storybook I got the following error using Firefox: 

```
Error: Cannot find module '@storybook/global'
    webpackEmptyContext sync:2
    js index.js:23
    Angular 2
    Webpack 25
[runtime.js:4:101929](http://localhost:6006/sb-preview/runtime.js)
    error runtime.js:4
    renderPreviewEntryError runtime.js:89
    renderPreviewEntryError runtime.js:103
    getProjectAnnotationsOrRenderError runtime.js:87
    _runRejections runtime.js:4
    _runRejections runtime.js:4
    catch runtime.js:4
    getProjectAnnotationsOrRenderError runtime.js:87
    initialize runtime.js:81
    js storybook-config-entry.js:25
```

Same error but using Chrome which provides a more explicit stack-trace:

```
   runtime.js:4 Error: Cannot find module '@storybook/global'
    at webpackEmptyContext ( sync:2:1)
    at ../../lib/channels/dist/index.js (index.js:23:1)
    at index.js:8:1
    at ./node_modules/@storybook/addon-actions/dist/index.js (index.js:23:1)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)
    at ./node_modules/storybook-addon-angular-router/dist/actionLoggingRouter.js (actionLoggingRouter.js:7:1)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)
    at ./node_modules/storybook-addon-angular-router/dist/withAngularRouter.js (withAngularRouter.js:10:1
```

    
To reproduce:

1. `npx @angular/cli@16.2 new mylib --no-create-application`
2. `cd mylib/`
3. `npx @angular/cli@16.2 generate library my-lib`
4. `npx storybook@latest init`
5. `npm i -D storybook-addon-angular-router`
6. Add the storybook-addon-angular-router addon in `/mylib/projects/my-lib/.storybook/main.ts`
7. `ng run my-lib:storybook`
8. Access http://localhost:6006/
9. F12 to show JS console and check the error

--- a/scripts/prepare/addon-bundle.ts
+++ b/scripts/prepare/addon-bundle.ts
@@ -120,7 +120,7 @@ const run = async ({ cwd, flags }: { cwd: string; flags: string[] }) => {
         ...(optimized ? dtsConfig : {}),
         entry: exportEntries,
         format: ['cjs'],
-        target: 'chrome100',
+        target: browserOptions.target,
         platform: 'neutral',
         external: [...commonExternals, ...globalManagerPackages, ...globalPreviewPackages],
         esbuildOptions: (options) => {

--- a/scripts/prepare/addon-bundle.ts
+++ b/scripts/prepare/addon-bundle.ts
@@ -120,13 +120,12 @@ const run = async ({ cwd, flags }: { cwd: string; flags: string[] }) => {
         ...(optimized ? dtsConfig : {}),
         entry: exportEntries,
         format: ['cjs'],
-        target: 'node18',
-        platform: 'node',
-        external: commonExternals,
+        target: 'chrome100',
+        platform: 'neutral',
+        external: [...commonExternals, ...globalManagerPackages, ...globalPreviewPackages],
         esbuildOptions: (options) => {
           /* eslint-disable no-param-reassign */
-          options.conditions = ['module'];
-          options.platform = 'node';
+          options.platform = 'neutral';
           Object.assign(options, getESBuildOptions(optimized));
           /* eslint-enable no-param-reassign */
         },

